Lib/distutils/sysconfig.py

```diff
@@ -205,7 +205,7 @@ def customize_compiler(compiler):
         if 'LDFLAGS' in os.environ:
             ldshared = ldshared + ' ' + os.environ['LDFLAGS']
         if 'CFLAGS' in os.environ:
-            cflags = opt + ' ' + os.environ['CFLAGS']
+            cflags = cflags + ' ' + os.environ['CFLAGS']
```

Comment:
I believe `opt` should be retained since there are compiler flags captured during `./configure` which would get dropped when the `CFLAGS` environment variable is set invoking a build via `distutils`.

For example, on my macOS machine, the following compiler flags would be lost if `CFLAGS` was set:

```pycon
>>> sysconfig.get_config_var('OPT')
'-DNDEBUG -g -fwrapv -O3 -Wall'
```

To the best of my understanding, it appears the goal is to have the compiler flags used when building the interpreter to be passed along to C/C++ extensions with the ability to add additional user-specified flags via the `CFLAGS` environment variable.

Given that the intent appears to be appending to the flags already captured during the configuration phase of building the interpreter, I would suggest:

```suggestion
            cflags = opt + ' ' + cflags + ' ' + os.environ['CFLAGS']
```

If the propose change is accepted, then my follow-up feedback would be updating the commit message  and news announcement accordingly.

Reply:
sysconfig reads PY_FLAGS from Makefile and expose it as "CFLAGS". In the Makefile, PY_CFLAGS is composed from other variables:

```
PY_CFLAGS=	$(BASECFLAGS) $(OPT) $(CONFIGURE_CFLAGS) $(CFLAGS) $(EXTRA_CFLAGS)
```

So "OPT" sysconfig variable is a subset of "CFLAGS" sysconfig variable.

Note: I'm talking about the specific case of sysconfig variables.

Python compiled manually:

```
$ ./python
Python 3.8.0a2+ (heads/common_config2-dirty:3fe7fa316f, Mar 14 2019, 23:17:45) 
>>> import shlex, sysconfig
>>> cflags = set(shlex.split(sysconfig.get_config_var('CFLAGS')))
>>> opt = set(shlex.split(sysconfig.get_config_var('OPT')))
>>> opt.issubset(cflags)
True
>>> cflags - opt
{'-Wno-unused-result', '-O0', '-Wsign-compare'}
>>> opt - cflags
set()
```

/usr/bin/python3:

```
vstinner@apu$ python3
Python 3.7.2 (default, Jan 16 2019, 19:49:22) 
>>> import shlex, sysconfig
>>> cflags = set(shlex.split(sysconfig.get_config_var('CFLAGS')))
>>> opt = set(shlex.split(sysconfig.get_config_var('OPT')))
>>> opt.issubset(cflags)
True
>>> cflags - opt
{'-Wsign-compare', '-Wno-unused-result'}
>>> opt - cflags
set()
```

Reply:
@vstinner, thanks for the walkthrough and have arrived at the same conclusion.